iaso/models/org_unit.py

```diff
@@ -861,11 +861,20 @@ def __apply_changes(self, user: User, approved_fields: typing.List[str]) -> None
                 self.org_unit.groups.add(*self.new_groups.all())
             elif field_name == "new_reference_instances":
                 self.org_unit.reference_instances.clear()
-                new_reference_instances = [
+
+                new_reference_instances = list(self.new_reference_instances.all())
+                new_reference_forms_ids = self.new_reference_instances.values_list("form_id", flat=True)
+
+                for instance in self.old_reference_instances.all():
```

Comment:
This is going to resurface old instances that have been changed in the mean time